micro-rdk/src/common/data_store.rs

```diff
@@ -70,6 +70,13 @@ pub trait DataStore {
         &mut self,
         collector_key: &ResourceMethodKey,
     ) -> Result<BytesMut, DataStoreError>;
+
+    /// Initializes from resource-method keys with a separate memory region dedicated to each key.
```

Comment:
- The part of the doc comment about memory regions seems too implementation focused for trait-level documentation.
- I'm a little unclear on why the trait should be mandating the form of construction for implementations.

Reply:
I think the concept of a data store only makes sense in the context of data management, which will always require that the store be configurable based on the collectors, so this doesn't feel like a very strict mandate. I do agree about taking out the "memory regions" part of the doc comment though